pkg/controller/operators/catalog/subscription/reconciler.go

```diff
@@ -84,7 +89,14 @@ func (c *catalogHealthReconciler) Reconcile(ctx context.Context, in kubestate.St
 					break
 				}
 
+				// TODO: reduce this to one update call
 				next, err = s.UpdateHealth(c.now(), c.client.OperatorsV1alpha1().Subscriptions(ns), catalogHealth...)
+				if err != nil {
+					return out, err
+				}
+				if c.sourceProvider != nil {
+					err = updateDeprecatedStatus(s.Subscription(), c.now(), c.client.OperatorsV1alpha1().Subscriptions(ns), c.sourceProvider.Sources(s.Subscription().GetNamespace()))
```

Comment:
Reuse the `ns` variable here as well?
```suggestion
					err = updateDeprecatedStatus(s.Subscription(), c.now(), c.client.OperatorsV1alpha1().Subscriptions(ns), c.sourceProvider.Sources(ns))
```